iphone/Maps/Classes/Components/ExpandableTextView/ExpandableTextView.swift

```diff
@@ -1,184 +1,120 @@
-      truncate = size.height < fullSize.height
-      if truncate {
-        let expandSize = expandText.size(width: frame.width, font: textFont, maxNumberOfLines: 1)
-        let layer = CATextLayer()
-        layer.position = CGPoint(x: 0, y: size.height)
-        layer.bounds = CGRect(origin: CGPoint(), size: expandSize)
-        layer.anchorPoint = CGPoint()
-        layer.string = expandText
-        layer.font = CGFont(textFont.fontName as CFString)
-        layer.fontSize = textFont.pointSize
-        layer.foregroundColor = expandTextColor?.cgColor
-        layer.contentsScale = UIScreen.main.scale
-
-        self.layer.addSublayer(layer)
-      }
-    } else {
-      size = fullSize
-    }
-
-    let layer = CATextLayer()
-    layer.bounds = CGRect(origin: CGPoint(), size: size)
-    layer.anchorPoint = CGPoint()
-    layer.string = text
-    layer.isWrapped = true
-    layer.truncationMode = truncate ? kCATruncationEnd : kCATruncationNone
-    layer.font = CGFont(textFont.fontName as CFString)
-    layer.fontSize = textFont.pointSize
-    layer.foregroundColor = textColor?.cgColor
-    layer.contentsScale = UIScreen.main.scale
-
-    self.layer.addSublayer(layer)
-  }
-
-  public override func awakeFromNib() {
-    super.awakeFromNib()
-    setup()
-  }
-
-  public convenience init() {
-    self.init(frame: CGRect())
-  }
-
-  public override init(frame: CGRect) {
+final class ExpandableReviewView: UIView {
+  var contentLabel: UILabel = {
+    let label = UILabel(frame: .zero)
+    label.numberOfLines = 0
+    label.clipsToBounds = true
+    label.translatesAutoresizingMaskIntoConstraints = false
+    return label
+  }()
+  
+  var moreLabel: UILabel = {
+    let label = UILabel(frame: .zero)
+    label.clipsToBounds = true
+    label.translatesAutoresizingMaskIntoConstraints = false
+    return label
+  }()
+  
+  var moreZeroHeight: NSLayoutConstraint!
+  private var settings: ExpandableReviewSettings = ExpandableReviewSettings()
+  private var isExpanded = false
+  private var onUpdateHandler: (() -> Void)?
+  
+  override init(frame: CGRect) {
     super.init(frame: frame)
-    setup()
+    configureContent()
   }
-
-  public required init?(coder aDecoder: NSCoder) {
+  
+  required init?(coder aDecoder: NSCoder) {
     super.init(coder: aDecoder)
-    setup()
-  }
-
-  override func mwm_refreshUI() {
-    super.mwm_refreshUI()
-    textColor = textColor?.opposite()
-    expandTextColor = expandTextColor?.opposite()
-  }
-
-  private func setup() {
+    configureContent()
+  }
+  
+  override func layoutSubviews() {
+    super.layoutSubviews()
+    updateRepresentation()
+  }
+  
+  func configureContent() {
+    self.addSubview(contentLabel)
+    self.addSubview(moreLabel)
+    let labels: [String: Any] = ["contentLabel": contentLabel, "moreLabel": moreLabel]
+    var contentConstraints: [NSLayoutConstraint] = []
+    let verticalConstraints = NSLayoutConstraint.constraints(withVisualFormat: "V:|-0-[contentLabel]-0-[moreLabel]",
```

Comment:
Is there any reason to specify zero spacing `-0-` explicitly?

Reply:
Yes, otherwise it will apply the default margins (8pt for our case).